### PR TITLE
Return a Result from Taffy::remove

### PR DESCRIPTION
**Pull request by plof27**
# Objective

Fixes: #102 

Returning a `Result` here has two nice benefits for users:
- The API is more consistent with similar methods like `Sprawl::remove_child` and `Sprawl::remove_child_at_index`
- The end user can verify that the removal actually removed something, and what that something actually was

## Context

The original issue suggests using an `Option<Id>` for this, but I decided to use `Result<usize, Error>` for three reasons:
- Better consistency with other API methods (see above)
- `Sprawl::find_node` returns a `Result`, and it would be good to propagate that error up.
- `NodeId` is only used internally, so the public API should use the underlying `usize`

I also updated the docstring, as returning a `usize` without any additional context could be confusing.

## Feedback wanted

~~This is a breaking change. I understand I'm supposed to update `RELEASES.md`, but the exact details on this aren't super clear. Is what I wrote acceptable?~~ Resolved

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 - renamed crate from `stretch2` to `sprawl`
 - updated to the latest version of all dependencies to reduce upstream pain caused by duplicate dependencies
 - renamed `stretch::node::Strech` -> `sprawl::node::Sprawl`
+- `Sprawl::remove` now returns a `Result<usize, Error>`, to indicate if the operation was sucessful, and if it was, which ID was invalidated.
```
Comment (alice-i-cecile) [inline RELEASES.md:11]: This should be under a 0.2.0 heading, as I don't want to publish this in 0.1 to ease the migration from stretch.

Comment (plof27) [reply RELEASES.md:11]: Ah, right. Thanks!

File: src/node.rs [hunk 2]
```diff
@@ -97,9 +97,9 @@ impl Sprawl {
         self.forest.clear();
     }
 
-    /// Remove nodes.
-    pub fn remove(&mut self, node: Node) {
-        let id = if let Ok(id) = self.find_node(node) { id } else { return };
+    /// Remove nodes. Returns the id of the node removed.
```
Comment (alice-i-cecile) [inline src/node.rs:100]: Docs could be better, but that's not your fault. If #96 is merged first, we'll need to rebase this.

Comment (plof27) [reply src/node.rs:100]: Fixed as of https://github.com/DioxusLabs/sprawl/pull/106/commits/8b227be403d33ea05dfe771b143821afd9e1a98e

File: RELEASES.md (hunk 1 above)
Comment (plof27) [reply RELEASES.md]: Fixed as of https://github.com/DioxusLabs/sprawl/pull/106/commits/8b227be403d33ea05dfe771b143821afd9e1a98e

File: src/node.rs (hunk 2 above)
Comment (plof27) [reply src/node.rs:100]: Good to know. I think we can cross that bridge when we come to it.

**Review by alice-i-cecile — APPROVED** [review]

**Review by Weibye — APPROVED** [review]
Looks good and makes sense :)

**Comment by Weibye** [comment]
PR title should be updated to "Return a Result from Taffy::remove"

File: tests/node.rs [hunk 3]
```diff
@@ -111,7 +111,7 @@ mod node {
         assert_eq!(taffy.children(node0).unwrap().as_slice(), &[node1]);
 
         // Disconnect the tree: <0> <2>
-        taffy.remove(node1);
+        let _ = taffy.remove(node1).unwrap();
```
Comment (alice-i-cecile) [inline tests/node.rs:114]:
```suggestion
        taffy.remove(node1).unwrap();
```

**Comment by alice-i-cecile** [comment]
@sixfold I broke the merge, sorry! Are you able to fix this?

**Comment by plof27** [comment]
> @sixfold I broke the merge, sorry! Are you able to fix this?

Probably. I'll take a look.

**Comment by plof27** [comment]
Should be all fixed now :)

**Pull request merged**